src/main/lrsql/util/oidc.clj

```diff
@@ -118,11 +151,28 @@
                            (not-empty sub)
                            (some-> scope
                                    (parse-scope-claim
-                                    :scope-prefix scope-prefix)
+                                    :scope-prefix scope-prefix
+                                    :scope-map oidc-scope-str-kw-map)
                                    not-empty
                                    (->> (into #{}))))]
         {:scopes      scopes
          :username    sub
          :oidc-issuer iss}
         ;; no valid scopes, can't do anything
         ::unauthorized))))
+
+(s/fdef authorize-admin-action
+  :args (s/cat :ctx           (s/keys :req-un [::auth/request])
+               :auth-identity (s/keys :req-un [::scopes]))
+  :ret (s/keys :req-un [::auth/result]))
+
+(defn authorize-admin-action
+  "Given a pedestal context and an OIDC admin auth identity, authorize or deny."
+  [{{:keys [request-method path-info]} :request
+    :as _ctx}
+   {:keys [scopes]
+    :as _auth-identity}]
+  {:result
+   (boolean
+    (and (cs/starts-with? path-info "/admin")
+         (contains? scopes :scope/admin)))})
```

Comment:
So you're basing the difference on the REST endpoint path, not anything fundamental about the OIDC token itself? (I'm assuming it's because the latter is impossible and that an OIDC token is an OIDC token.)

Reply:
Well `scope` is itself a claim on OIDC access tokens, for this exact case (the LRS/admin routes are a "resouce server"). So prior to this the scope claim is decoded and matched to known scopes, subject to the custom user prefix. I just used `:path-info` because we only have the one scope, in a more complex scenario it would probably use more than just that.

Reply:
For reference, when my OIDC client (which is just a modified version of the re-oidc demo rn) makes its initial auth request to the IDP, it specifies that it wants `all`, `admin`, etc. scopes, and that its "audience" includes the uri (known to the IDP) of the LRS. Then the IDP can (however it wants) decide to grant these or not. If it does, they are included in the token claims (and thus verifiable).